### PR TITLE
Redirect to the binary file in s3

### LINKED ISSUE #179: Add route to API spec for File (binary) GET
- add to API route for return a binary directly (doesn't exist currently; we have retrieve a File resource via /resource)

### LINKED ISSUE #180: Add File (binary) retrieval Handler
- check for permissions (stub or whatevs)
- return binary (from s3)
- return a filename in header
- return content disposition

### PR DESCRIPTION
**Pull request by jcoyne**
This will redirect to a pre-signed object URL in S3. This may be
preferable to the proxying approach, because it would limit the data
transfer that has to go through the taco server.
I believe this approach is preferable to #209 

Fixes #179
Fixes #180

### CODE REVIEW

**Comment by aaron-collier** [comment]
@jcoyne needs a quick rebase.

**Comment by jcoyne** [comment]
@aaron-collier rebase complete.

**Comment by jcoyne** [comment]
@aaron-collier Can you explain exactly what needs refactoring?

**Comment by aaron-collier** [comment]
@jcoyne I believe we thought there might be some changes needed after #245, but I could be mistaken. Lets review in the AM if @eefahy is available for standup.

File: handlers/deposit_file.go [hunk 1]
```diff
@@ -70,12 +70,16 @@ func (d *depositFileEntry) createFileResource(resourceID string, filename string
...
+	// TODO: What should be the default access for a file?
 	resource.Access = "private"
 	resource.AtContext = atContext
 	resource.AtType = fileType
 	resource.Label = filename
 	resource.Preserve = false
 	resource.Publish = false
+	// This will help future proof files in case at some point we decide that the
+	// resource identifier is not a suitable S3 key
+	resource.ContentKey = resourceID
```
Comment (aaron-collier) [inline handlers/deposit_file.go]: Do we need this? It seems like our ID should be sufficient.

File: handlers/deposit_file.go [hunk 2]
```diff
@@ -70,12 +70,16 @@ func (d *depositFileEntry) createFileResource(resourceID string, filename string
...
+	resource.ContentBucket = d.rt.Config().S3BucketName
```
Comment (eefahy) [inline handlers/deposit_file.go]: I don't think the changing uploading behavior or how we persist a resource is appropriate in this commit, nor does it reflect an agreed to change in the shape of a resource.

Comment (jcoyne) [reply handlers/deposit_file.go]: Prior to this commit we have no defined way to store where we persist the resource.  We need to store this data somewhere. Can you tell me what I ought to do?

**Pull request merged**